atom/index.js

```diff
@@ -24,28 +24,29 @@ export let atom = (initialValue, level) => {
         }
       }
     },
-    notify(changedKey) {
+    notify(oldVal, changedKey) {
       let runListenerQueue = !listenerQueue.length
       for (let i = 0; i < listeners.length; i += 2) {
         listenerQueue.push(
           listeners[i],
           listeners[i + 1],
           $atom.value,
-          changedKey,
+          oldVal,
+          changedKey
         )
       }
 
       if (runListenerQueue) {
         for (let i = 0; i < listenerQueue.length; i += 4) {
           let skip
-          for (let j = i + 1; !skip && (j += 4) < listenerQueue.length;) {
+          for (let j = i + 1; !skip && (j += 4) < listenerQueue.length; ) {
             if (listenerQueue[j] < listenerQueue[i + 1]) {
               skip = listenerQueue.push(
-               listenerQueue[i],
-               listenerQueue[i + 1],
-               listenerQueue[i + 2],
-               listenerQueue[i + 3]
-             )
+                listenerQueue[i],
+                listenerQueue[i + 1],
+                listenerQueue[i + 2],
+                listenerQueue[i + 3]
+              )
```

Comment:
the spaces here are not my design, they were added when running `eslint --fix` 🤷